phpBB/phpbb/console/command/user/activate.php

```diff
@@ -0,0 +1,221 @@
+	protected $log;
+
+	/** @var \phpbb\notification\manager */
+	protected $notifications;
+
+	/**
+	 * phpBB root path
+	 *
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * PHP extension.
+	 *
+	 * @var string
+	 */
+	protected $php_ext;
+
+	/**
+	 * Construct method
+	 *
+	 * @param \phpbb\user                       $user
+	 * @param \phpbb\db\driver\driver_interface $db
+	 * @param \phpbb\config\config              $config
+	 * @param \phpbb\language\language          $language
+	 * @param \phpbb\log\log_interface          $log
+	 * @param \phpbb\notification\manager       $notifications
+	 * @param string                            $phpbb_root_path
+	 * @param string                            $php_ext
+	 */
+	public function __construct(\phpbb\user $user, \phpbb\db\driver\driver_interface $db, \phpbb\config\config $config, \phpbb\language\language $language, \phpbb\log\log_interface $log, \phpbb\notification\manager $notifications, $phpbb_root_path, $php_ext)
+	{
+		$this->db = $db;
+		$this->config = $config;
+		$this->language = $language;
+		$this->log = $log;
+		$this->notifications = $notifications;
+		$this->phpbb_root_path = $phpbb_root_path;
+		$this->php_ext = $php_ext;
+
+		$this->language->add_lang('acp/users');
+		parent::__construct($user);
+	}
+
+	/**
+	 * Sets the command name and description
+	 *
+	 * @return null
+	 */
+	protected function configure()
+	{
+		$this
+			->setName('user:activate')
+			->setDescription($this->language->lang('CLI_DESCRIPTION_USER_ACTIVATE'))
+			->setHelp($this->language->lang('CLI_HELP_USER_ACTIVATE'))
+			->addArgument(
+				'username',
+				InputArgument::REQUIRED,
+				$this->language->lang('CLI_DESCRIPTION_USER_ACTIVATE_USERNAME')
+			)
+			->addOption(
+				'deactivate',
+				'd',
+				InputOption::VALUE_NONE,
+				$this->language->lang('CLI_DESCRIPTION_USER_ACTIVATE_DEACTIVATE')
+			)
+			->addOption(
+				'send-email',
+				null,
+				InputOption::VALUE_NONE,
+				$this->language->lang('CLI_DESCRIPTION_USER_ADD_OPTION_NOTIFY')
+			)
+		;
+	}
+
+	/**
+	 * Executes the command user:activate
+	 *
+	 * Deletes a user from the database. An option to delete the user's posts
+	 * is available, by default posts will be retained.
+	 *
+	 * @param InputInterface  $input  The input stream used to get the options
+	 * @param OutputInterface $output The output stream, used to print messages
+	 *
+	 * @return int 0 if all is well, 1 if any errors occurred
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output)
+	{
+		$io = new SymfonyStyle($input, $output);
+
+		$name = $input->getArgument('username');
+		$mode = ($input->getOption('deactivate')) ? 'deactivate' : 'activate';
+
+		if (!$user_row = $this->get_user_data($name))
+		{
+			$io->error($this->language->lang('NO_USER'));
+			return 1;
+		}
+
+		// Check if the user is already active (or inactive)
+		if ($mode == 'activate' && $user_row['user_type'] != USER_INACTIVE)
```

Comment:
Well we're not dealing with any variables or user input here. `$mode` will always be a string, either activate or deactivate, and nothing else, since it's essentially hard-coded up on line 126.